Make CLI Commands more independent

JordanShurmer opened the pull request:
**Note:** this PR is to the `cli-refactor` branch, not master.

# What Changed

* The commands in `cli/` are more independent and self-actualizing
  * They export their own usage string
  * They parse their own command line options
* cli.js is cleaned up
  * it only parses the options that it cares about
  * handles printing usage info for every command
  * exports the current version

# I Need Help

I'm not sure how to implement the `run` command. This PR adds it as an `unimplemented()` piece of code. 
We could merge this into the `cli-refactor` branch then fix the run command in there, or we can update this PR if someone can assist in implementing it.

JordanShurmer commented:
Test it out here:

```
deno run --unstable -A https://raw.githubusercontent.com/JordanShurmer/lume/1872c101bd57a943908a5666daa6339da410cdb1/cli.js --help


deno run --unstable -A https://raw.githubusercontent.com/JordanShurmer/lume/1872c101bd57a943908a5666daa6339da410cdb1/cli.js build --help

# etc.
```

oscarotero commented:
Hi, thanks for your work here.
Both `build` and `run` need an instance of `site`. This is why in my implementation I've created the site before executing these two commands ([See here](https://github.com/lumeland/lume/blob/feature/cli-refactor/cli.js#L83)).

Maybe cli.js can export a `createSite(options)` function that can be imported and used by those two commands (like you did with `version`).

Then, `run` only need to execute [`site.run()`](https://github.com/lumeland/lume/blob/feature/cli-refactor/cli.js#L116) with the name of the script. For example: `lume run deploy` should execute `site.run("deploy")`.

Another thing: can you run `deno fmt` on finish this pull request? I want to use the same code format of Deno (the only exception is CHANGELOG.md) that should be ignored.

Thanks!

oscarotero commented on cli.js:
```diff
@@ -1,124 +1,146 @@
 
 if (import.meta.main) {
-  cli(Deno.args);
+  await cli(Deno.args);
```
Is this `await` needed?

JordanShurmer replied:
I did have this in a `try...catch` at one point, to log error messages better. Not sure why I removed the `try...catch` 🤔 

WDYT? worth catching all errors and formatting a prettier error message, or just leave it and let the normal Deno error logging show?

oscarotero replied:
I have an utility function to show errors (https://github.com/lumeland/lume/blob/master/utils.js#L72, with much room for improvement).
A prettier error message whould be fine, errors in Deno aren't very friendly.

oscarotero commented on cli.js:
```diff
@@ -1,124 +1,146 @@
     return;
   }
 
   if (options._.length > 1) {
```
We have to remove this from there, because `run` command need two values here: `lume run deploy`. I think this should be checked by every command individually (like the arguments).

JordanShurmer replied:
done

oscarotero commented on cli.js:
```diff
@@ -1,124 +1,146 @@
+    let usage;
+    if (command) {
+      const commandModule = await import(`./cli/${command}.js`).catch(_ => {
+      }) //ignore import errors here
```
Why ignore the errors here?

oscarotero replied:
Maybe, if we import the commands as `import * as build from "./cli/build.js"`, we don't need to import again here, just run `build.USAGE` (that i'd rename to `help`, for coherence with the option name).

shadowtime2000 replied:
I think the only time an error would occur is if the command doesn't exist and in that case we should tell them.

JordanShurmer replied:
I'm ignoring it here because it would mean the command doesn't exist, which is a scenario handled at the end of the command checks. 

@oscarotero I'll play around with other ways to import, to try to avoid this dynamic import

JordanShurmer commented:
Thanks. all, for reviewing.

Made various changes to address the feedback:

* Implemented the run command
  * pulled the site building logic into a shared function in cliUtils.js
* Refactored the way cli.js pulls in the commands to avoid doing dynamic import
* Tried to clean up cli.js even more, let me know what you think
* `deno fmt`

JordanShurmer commented:
Test out the latest here: https://raw.githubusercontent.com/JordanShurmer/lume/1336e460d3220489fcfa50060290612795652d9a/cli.js

```
deno run --unstable -A https://raw.githubusercontent.com/JordanShurmer/lume/1336e460d3220489fcfa50060290612795652d9a/cli.js --help
```

oscarotero commented:
Great job. Thank you @JordanShurmer
